pandas/tests/io/test_http_auth.py

```diff
@@ -0,0 +1,213 @@
+import pytest
+import pandas as pd
+import pandas.util.testing as tm
+
+if pd.io.common.is_requests_pkg_avail():
+    from requests.packages.urllib3.exceptions import InsecureRequestWarning
+    import requests
+    requests_pkg = True
+else:
+    requests_pkg = False
+
+
+def _skip_if_no_requests():
+    if not requests_pkg:
+        pytest.skip('python-requests not installed, skipping')
+
+
+uname = 'pandasusr'
```

Comment:
Create fixtures out of these e.g.:

~~~python
@pytest.fixture
def username():
   return "pandasusr"
~~~

This is more idiomatic with `pytest`.  You can read more here:

https://docs.pytest.org/en/latest/fixture.html